nixos/modules/security/doas.nix

```diff
@@ -0,0 +1,266 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.security.doas;
+
+  inherit (pkgs) doas;
+
+  mkUsrString = user: toString user;
+
+  mkGrpString = group: ":${toString group}";
+in
+{
+
+  ###### interface
+
+  options.security.doas = {
+
+    enable = mkOption {
+      type = with types; bool;
+      default = true;
+      description = ''
+        Whether to enable the <command>doas</command> command, which
+        allows non-root users to execute commands as root.
+      '';
+    };
+
+    wheelNeedsPassword = mkOption {
+      type = with types; bool;
+      default = true;
+      description = ''
+        Whether users of the <code>wheel</code> group must
+        provide a password to run commands as super user via <command>doas</command>.
+      '';
+    };
+
+    configFile = mkOption {
```

Comment:
I'm very confused by the use of both `extraConfig` & `configFile` and how they interact.

Is there really a point to `configFile` when someone not wanting the configuration provided by this module can set `environment.etc."doas.conf"`?

Reply:
The simplest answer is: the `sudo` module had both, and all I did was copy-paste the module and apply liberal changes to it. So, they'll interact the same way that `sudo`'s `extraConfig` and `configFile` do.


I think it would be nice to be mostly uniform with the `sudo` module so it's *almost* a drop-in replacement, but I don't have strong feelings about it. Feel free to sway me :)

Reply:
[We conversed on IRC](https://logs.nix.samueldr.com/nixos/2020-05-02#1588446280-1588446713;) and I've decided to drop this. The reason `sudo` appears to have this  is mostly historical/for back-compat. Since this is brand-spanking-new, there's no reason to carry around historical baggage from elsewhere :)